### PR TITLE
Tests for tuples

### PR DESCRIPTION
**Pull request by prcastro**
As suggested by #9493, I made some tests for tuples, trying to improve coverage of functions associated with them. I added a new file just to test `base/tuple.jl`, including tests for almost all functions on that file.

I hope it helps. =)

### CODE REVIEW

**Comment by nalimilan** [comment]
Thanks. I've restarted the builds as the failures seem unrelated.

**Comment by ViralBShah** [comment]
That's a nice set of tests!

File: test/tuple.jl [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+@test size((1,2), 1) === 2
+
+# @test_throws ArgumentError size((), 2)
+# @test_throws ArgumentError size((1,), 2)
+# @test_throws ArgumentError size((1,2), 2)
+
+@test getindex((1,), 1) === 1
+@test getindex((1,2), 2) === 2
+@test_throws BoundsError getindex((), 1)
+@test_throws BoundsError getindex((1,2), 0)
```
Comment (hayd) [inline test/tuple.jl]: could also test for negative indexing here (is BoundsError) :)

Comment (brk00) [inline test/tuple.jl]: Just did it :)

**Comment by brk00** [comment]
(Off Topic: When you squash commits locally, how do you push it to your repo? I used `git push --force origin master`, but I'm now wure if it's the usual way to do it)

**Comment by hayd** [comment]
@brk00 You have to force push to the branch to "change history", which rebase does. As you can see only one commit appears here now. This is the correct/only way to do it!

A git tip, though don't worry about it this time, is to push your commits to a named branch (one which isn't named master!) e.g. "tuple_tests". In git branching is really cheap, and you can/should branch (from master) for each piece of work - this allows your branches to be independent. This also keeps your master clean: you don't want to start working on a slightly different master to upstream/everyone else.

**Comment by brk00** [comment]
@hayd Thanks for the tips! I'll follow these guidelines from now on. =) (It would also be nice to have a section on CONTRIBUTING.md explaining this sort of operation)

**Comment by timholy** [comment]
Good point. There is a potentially-useful set of tips in a related section on [Packages](http://docs.julialang.org/en/latest/manual/packages/#squashing-and-rebasing).

**Comment by brk00** [comment]
How could I miss this? Thanks.

**Comment by timholy** [comment]
Well, it was added recently. I don't keep up well with changes to documentation I first read long ago.
